Версия № 5:

ZotovSlava opened the pull request:
1) Добавлена новая реализация класса для работы с памятью FileBackedTaskManager extends InMemoryTaskManager новый класс позволяет записывать изменения в файл и загружать его при запуске программы;

2) Добавлен Тест нового класса FileBackedTaskManagerTest;

3) Создано пользовательское исключение ManagerSaveException;

4) Добавлен enum TaskType.

ZotovSlava committed:
Версия № 5:

1) Добавлена новая реализация класса для работы с памятью FileBackedTaskManager extends InMemoryTaskManager новый класс позволяет записывать изменения в файл и загружать его при запуске программы;

2) Добавлен Тест нового класса FileBackedTaskManagerTest;

3) Создано пользовательское исключение ManagerSaveException;

4) Добавлен enum TaskType.

ZotovSlava committed:
Версия № 5.1:

1) Исправления замечаний Test GitHub

ZotovSlava committed:
Версия № 5.2:

1) Исправления замечаний Test GitHub

ZotovSlava committed:
Версия № 5.3:

1) Корректировка тестов.

litvitnik commented on src/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    private static void loadFromFile(File file) {
```
Такого метода быть не должно. По ТЗ нужно создать метод public static FileBackedTaskManager loadFromFile(File file), который будет восстанавливать данные менеджера из файла при запуске программы.
Также нужно создать метод Task fromString(String value), он должен быть приватным

litvitnik commented on test/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,87 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FileBackedTaskManagerTest {
```
Суть тестов должна быть в том, что мы  должны создать менеджер конструктором, повыполнять разные действия с задачами в нем, а затем создать второй из того же файла через FileBackedTaskManager.loadFromFile(String filename), и убедиться, что их содержимое равно

ZotovSlava committed:
Версия № 5.4:

1) Внесны изменения в класс FileBackedTaskManager
- изменен метод loadFromFile;
- добавлен метод fromString, который забрал часть функционала loadFromFile;

2) Тестовый класс FileBackedTaskManagerTest переработан под новую реализацию класса FileBackedTaskManager.

ZotovSlava committed:
Версия № 5.5:
Исправление для GitTest
